cloudinit/sources/DataSourceAzure.py

```diff
@@ -1210,6 +1207,16 @@ def exc_cb(msg, exception):
 
         return return_val
 
+    @azure_ds_telemetry_reporter
+    def _get_reprovision_data_from_imds(
+            self, url, headers, exc_cb):
+        return readurl(url,
+                       timeout=IMDS_TIMEOUT_IN_SECONDS,
+                       headers=headers,
+                       exception_cb=exc_cb,
+                       infinite=True,
+                       log_req_resp=False).contents
+
```

Comment:
@anhvoms This [unit test](https://github.com/canonical/cloud-init/blob/d95b448fe106146b7510f7b64f2e83c51943f04d/tests/unittests/test_datasource/test_azure.py#L2965-L2972) (already in cloud-init master from previous work) ensures that `readurl` is called with the correct args during `_poll_imds`.